src/libraries/System.Private.CoreLib/src/System/Collections/Generic/HashSet.cs

```diff
@@ -0,0 +1,1564 @@
+                    else
+                    {
+                        // Object type: Shared Generic, EqualityComparer<TValue>.Default won't devirtualize (https://github.com/dotnet/runtime/issues/10050),
+                        // so cache in a local rather than get EqualityComparer per loop iteration.
+                        EqualityComparer<T> defaultComparer = EqualityComparer<T>.Default;
+                        int i = GetBucketValue(hashCode) - 1; // Value in _buckets is 1-based
+                        while (i >= 0)
+                        {
+                            ref Entry entry = ref entries[i];
+                            if (entry.HashCode == hashCode && defaultComparer.Equals(entry.Value, item))
+                            {
+                                return i;
+                            }
+                            i = entry.Next;
+
+                            collisionCount++;
+                            if (collisionCount > (uint)entries.Length)
+                            {
+                                // The chain of entries forms a loop, which means a concurrent update has happened.
+                                ThrowHelper.ThrowInvalidOperationException_ConcurrentOperationsNotSupported();
+                            }
+                        }
+                    }
+                }
+                else
+                {
+                    int hashCode = item != null ? comparer.GetHashCode(item) : 0;
+                    int i = GetBucketValue(hashCode) - 1; // Value in _buckets is 1-based
+                    while (i >= 0)
+                    {
+                        ref Entry entry = ref entries[i];
+                        if (entry.HashCode == hashCode && comparer.Equals(entry.Value, item))
+                        {
+                            return i;
+                        }
+                        i = entry.Next;
+
+                        collisionCount++;
+                        if (collisionCount > (uint)entries.Length)
+                        {
+                            // The chain of entries forms a loop, which means a concurrent update has happened.
+                            ThrowHelper.ThrowInvalidOperationException_ConcurrentOperationsNotSupported();
+                        }
+                    }
+                }
+            }
+
+            return -1;
+        }
+
+        /// <summary>Gets the index in the bucket corresponding to the specified hashcode.</summary>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private int GetBucketValue(int hashCode)
+        {
+            int[] buckets = _buckets!;
+#if TARGET_64BIT
+            return buckets[HashHelpers.FastMod((uint)hashCode, (uint)buckets.Length, _fastModMultiplier)];
+#else
+            return buckets[(uint)hashCode % (uint)buckets.Length];
+#endif
+        }
+
+        /// <summary>Gets the index in the bucket corresponding to the specified hashcode.</summary>
```

Comment:
I think it gets the bucket (an element of `_buckets`), not the index.

Reply:
It's an index into _entries.

Reply:
> It's an index into _entries.

I see.  My confusion ("index in the bucket" vs. "index into _entries") could be avoided with slight rewording of the comment: "Gets the index **stored** in the bucket" or "Gets the **_entry** index in the bucket".

Reply:
Yup. Already reworded it locally, just haven't pushed it up yet.